dione-spark/src/main/scala/com/paypal/dione/spark/avro/btree/SparkAvroBtreeUtils.scala

```diff
@@ -136,41 +136,34 @@ object SparkAvroBtreeUtils {
       else {
         val bufIt = it.buffered
         val row = bufIt.head
-        val folder = tableLocationStr + "/" + partitionKeys.map(pk => (pk, row.getAs[String](pk))).map(p => p._1 + "=" + p._2).mkString("/")
-        val fs = new Path(folder).getFileSystem(new Configuration)
-        if (!fs.exists(new Path(folder)))
+        val folder = tableLocationStr + "/" + getFolderLocationFromRow(row, partitionKeys)
+        if (!checkIfFolderExists(folder))
           Iterator.empty
         else {
 
-          val avroBtreeStorageFileReader = AvroHashBtreeStorageFolderReader(folder).getFile(keys.map(row.getAs[Any]))
+          val avroBtreeStorageFileReader = AvroHashBtreeStorageFolderReader(folder).getFile(row.getAs[Int](KEY_HASH_COLUMN))
           val kvIt = avroBtreeStorageFileReader.getIterator().buffered
           val converter = AvroSerializerHelper.avroDeserializer(avroBtreeStorageFileReader.fileReader.getValueSchema, indexTableValueSchema)
 
           bufIt.flatMap(row => {
             val key1GR = avroBtreeStorageFileReader.toGR(keys.map(row.getAs[Any]))
-            var cmp = -100
-            while (cmp<0 && kvIt.hasNext) {
-              val head = kvIt.head
-              cmp = GenericData.get.compare(head._1, key1GR, avroBtreeStorageFileReader.fileReader.getKeySchema)
-              logger.debug("comparing file key {} with DS key {} and got {}", head._1, key1GR, cmp+"")
-              if (cmp < 0)
-                kvIt.next()
-            }
-            if (!kvIt.hasNext || cmp>0)
+            val a = avroBtreeStorageFileReader.get(key1GR)
```

Comment:
not sure that this is the correct baseline to compare, as I have a feeling that it will be much worse then just reading through all the index file (as was before). Even now after fixing #73 it should be much faster.